Python/pylifecycle.c

```diff
@@ -345,103 +420,157 @@ _Py_InitializeEx_Private(int install_sigs, int install_importlib)
         Py_LegacyWindowsStdioFlag = add_flag(Py_LegacyWindowsStdioFlag, p);
 #endif
 
-    _Py_HashRandomization_Init();
+    _INIT_DEBUG_PRINT("INITIALISING HASH RANDOMISATION\n");
+    _Py_HashRandomization_Init(&core_config);
+    if (!core_config.use_hash_seed || core_config.hash_seed) {
+        /* Random or non-zero hash seed */
+        Py_HashRandomizationFlag = 1;
+    }
 
+    _INIT_DEBUG_PRINT("MAKING FIRST INTERPRETER\n");
     _PyInterpreterState_Init();
     interp = PyInterpreterState_New();
     if (interp == NULL)
-        Py_FatalError("Py_Initialize: can't make first interpreter");
+        Py_FatalError("Py_InitializeCore: can't make main interpreter");
+    interp->core_config = core_config;
 
+    _INIT_DEBUG_PRINT("MAKING FIRST THREAD STATE\n");
     tstate = PyThreadState_New(interp);
     if (tstate == NULL)
-        Py_FatalError("Py_Initialize: can't make first thread");
+        Py_FatalError("Py_InitializeCore: can't make first thread");
     (void) PyThreadState_Swap(tstate);
 
+    _INIT_DEBUG_PRINT("INITIALISING GIL STATE API\n");
 #ifdef WITH_THREAD
     /* We can't call _PyEval_FiniThreads() in Py_FinalizeEx because
-       destroying the GIL might fail when it is being referenced from
-       another running thread (see issue #9901).
-       Instead we destroy the previously created GIL here, which ensures
        that we can call Py_Initialize / Py_FinalizeEx multiple times. */
-    _PyEval_FiniThreads();
-
```

Comment:
Almost certainly the result of a merge glitch somewhere along the line when I was still maintaining the patch out of tree.